packages/cli/src/config/extensions/extensionSettings.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+import { ExtensionStorage } from '../extension.js';
+import type { ExtensionConfig } from '../extension.js';
+
+import prompts from 'prompts';
+
+export interface ExtensionSetting {
+  name: string;
+  description: string;
+  envVar: string;
+  sensitive?: boolean;
+}
+
+export async function maybePromptForSettings(
+  extensionConfig: ExtensionConfig,
+  requestSetting: (setting: ExtensionSetting) => Promise<string>,
+): Promise<void> {
+  const { settings, name: extensionName } = extensionConfig;
+  if (!settings || settings.length === 0) {
+    return;
+  }
+
+  const envFilePath = path.join(
+    new ExtensionStorage(extensionName).getExtensionDir(),
+    '.env',
+  );
+  let envContent = '';
+
+  console.log(''); // for spacing
```

Comment:
This is baking in an assumption that we are not in the CLI - we wouldn't want to do this there.

Maybe the codepath that uses `readline` should do this?